test/src/specs/mining/fee.rs

```diff
@@ -1 +1,355 @@
+use crate::utils::generate_utxo_set;
+use crate::{Net, Node, Spec};
+use ckb_types::core::{BlockNumber, BlockView, Capacity, TransactionView};
+use ckb_types::packed::{Byte32, OutPoint};
+use ckb_types::prelude::Entity;
+use ckb_types::prelude::*;
+use std::collections::{HashMap, HashSet};
 
+pub struct FeeOfTransaction;
+
+impl Spec for FeeOfTransaction {
+    crate::name!("fee_of_transaction");
+
+    // Case: Only submit 1 transaction, and then wait for its proposed and committed
+    //
+    //   1. Submit transaction `tx` into transactions_pool after height `i`
+    //   2. Expect that the miner proposes `tx` within `block[i + 1]`
+    //   3. Expect that the miner commits `tx` within `block[i + 1 + PROPOSAL_WINDOW_CLOSEST]`
+    //   4. Expect that the miner receives the proposed reward of `tx` from
+    //      `block[i + 1 + FINALIZATION_DELAY_LENGTH]`
+    //   5. Expect that the miner receives the committed reward of `tx` from
+    //      `block[i + 1 + PROPOSAL_WINDOW_CLOSEST + FINALIZATION_DELAY_LENGTH]`
+    fn run(&self, net: Net) {
+        let node = &net.nodes[0];
+        let closest = node.consensus().tx_proposal_window().closest();
+        let finalization_delay_length = node.consensus().finalization_delay_length();
+
+        let txs = generate_utxo_set(node, 1).bang_random_fee(vec![node.always_success_cell_dep()]);
+        node.submit_transaction(&txs[0]);
+
+        let number_to_submit = node.get_tip_block_number();
+        let number_to_propose = number_to_submit + 1;
+        let number_to_commit = number_to_propose + closest;
+        node.generate_blocks(2 * finalization_delay_length as usize);
+        assert_block(node, number_to_propose, &txs, &[]);
```

Comment:
Suggest to use `assert_proposals`, `assert_committed` directly, `assert_block` is obscure. The same as `assert_chain`